File: 프로그래머스/lv1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.java
import java.util.Arrays;

class Solution {
    public int[] solution(int[] arr, int divisor) {
        int count = 0;
        Arrays.sort(arr);
        
        for(int i=0; i<arr.length; i++) {
            if(arr[i]%divisor==0) {
                count++;
            }
        }
        
        int[] answer = new int[count];
        int idx = 0;
        for(int i=0; i<arr.length; i++) {
            if(arr[i]%divisor==0) {
                answer[idx] = arr[i];
                idx++;
            }
        }
        
        if(answer.length == 0) {
            answer = new int[1];
            answer[0] = -1;
        }
        
        return answer;
    }
}

File: 프로그래머스/1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.js
function solution(arr, divisor) {
    const answer = [];
    for(e of arr) {
        if(e % divisor === 0) answer.push(e);
    }

    
    answer.length !== 0 ? answer.sort((a, b) => a - b) : answer.push(-1);
    
    return answer;
}